syntax = "proto3";
package minecpp.player;
import "game.proto";

message Data {
    string name = 1;
    int32 entity_id = 2;
    double x = 3;
    double y = 4;
    double z = 5;
    float yaw = 6;
    float pitch = 7;
    Abilities abilities = 8;
    RecipeBook recipe_book = 9;
    minecpp.game.GameMode game_mode = 10;
}

message Abilities {
    float walk_speed = 1;
    float fly_speed = 2;
    bool flying = 3;
    bool instant_build = 4;
    bool invulnerable = 5;
    bool may_build = 6;
    bool may_fly = 7;
}

message RecipeBook {
    bool gui_open = 1;
    bool filtering_craftable = 2;
    bool furnace_gui_open = 3;
    bool furnace_filtering_craftable = 4;
}